pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/exchange/SingletonExchange.java

```diff
@@ -38,9 +37,7 @@ class SingletonExchange extends BlockExchange {
   SingletonExchange(List<SendingMailbox> sendingMailboxes, BlockSplitter splitter,
       Function<List<SendingMailbox>, Integer> statsIndexChooser) {
     super(sendingMailboxes, splitter, statsIndexChooser);
-    Preconditions.checkArgument(
-        sendingMailboxes.size() == 1 && sendingMailboxes.get(0) instanceof InMemorySendingMailbox,
-        "Expect single InMemorySendingMailbox for SingletonExchange");
+    Preconditions.checkArgument(sendingMailboxes.size() == 1, "Expect single mailbox in Singleton Exchange");
```

Comment:
Do we need to change this? We don't want it to send data across servers. Currently singleton means local exchange.

Reply:
Singleton from a definition perspective means that there's a single receiver, and hence there should be only 1 sending mailbox. It doesn't really mean local. Data could be sent across different instances too (e.g. in lite mode we may send data to the broker from all of the servers, and all of the servers in that case would be using the SingletonExchange)

Our current Runtime is already **not** tying SingletonExchange with InMemorySendingMailbox. We have tied Singleton with Local only in the planner. With my changes we'll get rid of that too and use Singleton correctly.

Reply:
Yeah I understand we are using `SingletonExchange` as `LocalExchange` right now, which is not correct. My concern is that this change is loosing the check, and could potentially allow bug where data is sending across servers.

Do you think we can introduce a `LocalExchange` and modify the existing code to create that when `RelDistribution.Type` is `SINGLETON`? With your new code, we should be able to properly distinguish local (maybe never needed) vs singleton

Reply:
That will introduce more complications and will end up increasing the scope of incorrect semantics.

If we want to run validations like this they should be run as part of the planner itself. Someone from our team can work on adding that in a separate PR.

Reply:
This is not a blocker. We can revisit in the future